pkg/liquidity-mining/test/ChildChainGauge.test.ts

```diff
@@ -523,6 +523,88 @@ describe('ChildChainGauge', () => {
       });
     });
 
+    context('with broken rewards', () => {
+      let functionalReward: Contract;
+
+      sharedBeforeEach('add rewards and break token after deposit', async () => {
+        // Add and deposit broken reward
+        const brokenERC20 = await deploy('v2-solidity-utils/BreakableERC20Mock', { args: ['Broken token', 'BRK'] });
+        await vault.authorizerAdaptorEntrypoint
+          .connect(admin)
+          .performAction(
+            gauge.address,
+            gauge.interface.encodeFunctionData('add_reward', [brokenERC20.address, admin.address])
+          );
+        await brokenERC20.mint(admin.address, rewardAmount);
+        await brokenERC20.connect(admin).approve(gauge.address, rewardAmount);
+
+        await gauge.connect(admin).deposit_reward_token(brokenERC20.address, rewardAmount);
+        await brokenERC20.setIsBroken(true);
+
+        // Add and deposit functional reward
+        functionalReward = await deploy('v2-solidity-utils/TestToken', { args: ['Test functionalReward', 'RWRD', 18] });
+        await vault.authorizerAdaptorEntrypoint
+          .connect(admin)
+          .performAction(
+            gauge.address,
+            gauge.interface.encodeFunctionData('add_reward', [functionalReward.address, admin.address])
+          );
+        await functionalReward.mint(admin.address, rewardAmount);
+        await functionalReward.connect(admin).approve(gauge.address, rewardAmount);
+
+        await gauge.connect(admin).deposit_reward_token(functionalReward.address, rewardAmount);
+
+        // Only one user
+        await stakeBPT(bptAmount, bn(0));
+        await advanceToTimestamp((await currentTimestamp()).add(WEEK));
+        claimer = user1;
+      });
+
+      it('reverts claiming broken rewards with explicit index', async () => {
+        await expect(gauge.connect(claimer)[claim](claimer.address, ZERO_ADDRESS, [0])).to.be.revertedWith(
+          'BROKEN_TOKEN'
+        );
+      });
+
+      it('reverts claiming broken rewards with implicit indexes', async () => {
+        await expect(claimAll()).to.be.revertedWith('BROKEN_TOKEN');
+      });
+
+      it('can still claim working rewards', async () => {
+        const tx = await gauge.connect(claimer)[claim](claimer.address, ZERO_ADDRESS, [1]);
+        const event = expectTransferEvent(
+          await tx.wait(),
+          { from: gauge.address, to: claimer.address },
+          functionalReward.address
+        );
+
+        expect(event.args.value).to.be.almostEqual(rewardAmount);
+      });
```

Comment:
Can you add a test where we claim the working rewards, then fix the broken rewards, then reclaim, and

a) the broken rewards are claimed correctly
b) the already claimed rewards are not re-claimed

(i.e. claims are fully independent)

Reply:
https://github.com/balancer/balancer-v2-monorepo/pull/2325/commits/15ada0d0912fef81f13f6c11475c3f09d1563dcb there we go.